micropython_ota.py

```diff
@@ -46,28 +58,58 @@ def ota_update(host, project, filenames, use_version_prefix=True, user=None, pas
         if version_changed:
             try:
                 uos.mkdir('tmp')
-            except:
-                pass
+            except OSError as e:
+                if e.errno != 17:
+                    raise
+            if filenames is None:
+                filenames = fetch_manifest(host, project, remote_version, prefix_or_path_separator, auth=auth, timeout=timeout)
             for filename in filenames:
+                if filename.endswith('/'):
+                    dir_path="tmp"
```

Comment:
```suggestion
                    dir_path='tmp'
```

Reply:
We could define `dir_path` at the beginning of the method and reference this var in the `uos.mkdir` call.